nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java

```diff
@@ -433,7 +434,7 @@ public class CaptureChangeMySQL extends AbstractSessionFactoryProcessor {
     private BinlogLifecycleListener lifecycleListener;
     private GtidSet gtidSet;
 
-    private final LinkedBlockingQueue<RawBinlogEvent> queue = new LinkedBlockingQueue<>();
+    private final BlockingQueue<RawBinlogEvent> queue = new LinkedBlockingQueue<>(100);
```

Comment:
Is there a particular reason for selecting `100` as the queue size? Initial sets of change events could number in the thousands, but on the other hand, keeping this low avoids memory consumption issues.

Reply:
Since we're blocking waiting on the event to be added to the queue, @markap14 suggested we wouldn't need a very large buffer at all. I'm thinking maybe a larger buffer size would be for times when the processor is scheduled for a longer period than it should be? I welcome your and Mark's comments here, just went with his suggestion :)

Reply:
A large number of events can build up depending on the initial starting position in the binary log, or lots of activity. The value of `100` may still be best to avoid memory consumption, but I would have to take a closer look at the polling behavior to see whether there is value in a larger buffer to handle this kind of scenario.

Reply:
I'll make it `1000` to help handle bursts of events,  but since it's now a true blocking queue, we shouldn't miss events and the processor should be draining the buffer as fast as the library is populating it.